widget/textinput.go

```diff
@@ -426,9 +434,14 @@ func (t *TextInput) drawTextAndCaret(screen *ebiten.Image, def DeferredRenderFun
 	tr := rect
 	tr = tr.Add(img.Point{t.padding.Left, t.padding.Top})
 
+	inputStr := t.InputText
+	if t.secure {
+		inputStr = strings.Repeat("*", len([]rune(t.InputText)))
```

Comment:
This works like it should now.

Unfortunately, there's another problem I didn't catch right away. Since `drawTextAndCaret` is called for every GPU frame, `strings.Repeat` is as well, which is quite taxing on the GC. The result of `Repeat` should be cached as much as possible. I would be fine with an unexported field `secureInputText` or some such.

Reply:
Fixed!